core/jazz_slack-event-handler/index.js

```diff
@@ -0,0 +1,286 @@
+"use strict";
+
+const config = require("./components/config.js"); //Import the environment data.
+const logger = require("./components/logger.js"); //Import the logging module.
+const utils = require("./components/utils.js")(); //Import the utils module.
+const _ = require("lodash");
+const errorHandlerModule = require("./components/error-handler.js");
+const failureErrorCodes = require("./utils/failure-codes.js");
+const customErrorHandlerModule = require("./components/custom-error-handler.js"); //Import the custom error codes module.
+const format = require("string-template");
+
+const AWS = require("aws-sdk");
+const request = require("request");
+const rp = require('request-promise-native');
+
+var errorHandler = errorHandlerModule(logger);
+var failureCodes = failureErrorCodes();
+var retryErrorHandler = customErrorHandlerModule();
+var processedEvents = [];
+var failedEvents = [];
+
+var handler = (event, context, cb) => {
+    var configData = config(context);
+
+    logger.init(event, context);
+    logger.info("Event :" + JSON.stringify(event));
+
+    rp(getTokenRequest(configData))
+        .then(result => {
+            return getAuthResponse(result);
+        })
+        .then(authToken => {
+            return processRecords(event, configData, authToken);
+        })
+        .then(result => {
+            var records = getEventProcessStatus();
+            logger.info("Successfully processed events: " + JSON.stringify(records));
+            return cb(null, records);
+        })
+        .catch(err => {
+            var records = getEventProcessStatus();
+            logger.error("Error processing events: " + JSON.stringify(err));
+            return cb(null, records);
+        });
+}
+
+var getTokenRequest = (configData) => {
+    return {
+        uri: configData.SERVICE_API_URL + configData.TOKEN_URL,
+        method: 'post',
+        json: {
+            "username": configData.SERVICE_USER,
+            "password": configData.TOKEN_CREDS
+        },
+        rejectUnauthorized: false,
+        transform: (body, response, resolveWithFullResponse) => {
+            return response;
+        }
+    };
+};
+
+var getAuthResponse = (result) => {
+    return new Promise((resolve, reject) => {
+        if (result.statusCode === 200 && result.body && result.body.data) {
+            return resolve(result.body.data.token);
+        } else {
+            logger.error("getAuthResponse failed");
+            return reject(errorHandler.throwInternalServerError("Invalid token response from API"));
+        }
+    });
+}
+
+var processRecords = (event, configData, authToken) => {
+    return new Promise((resolve, reject) => {
+        var processRecordPromises = [];
+        for (var i = 0; i < event.Records.length; i++) {
```

Comment:
I had changed the runtime as 6.10 and rebuild, but the index itself not loading, seems much more syntax needs to be changed for runtime 6.10. Since its development already completed and tested, I am leaving this comments as of now and will taken care in a refactor story.

Reply:
what errors are you seeing?

Reply:
done